devtools/etdump/etdump_flatcc.cpp

```diff
@@ -497,27 +499,15 @@ ETDumpResult ETDumpGen::get_etdump_data() {
 }
 
 void ETDumpGen::set_debug_buffer(Span<uint8_t> buffer) {
```

Comment:
I think we should remove the `set_debug_buffer` api. Is there any reason to keep it around? Makes it a little confusing for users i think to have these 2 around.

Reply:
just for BC. Currently all our users use `set_debug_buffer` to set the memory for debug data, and their work will be impacted if we remove that

Reply:
Actually we should keep this API around i think. Only users who want to do their custom thing should know about data sinks, otherwise most users should just be able to call `set_debug_buffer`.

Reply:
User only need standard data dumping pipeline can use both datasink and `set_debug_buffer`, since we have provided `BufferDataSink` which is exactly the same as debug buffer pipeline. But from API design prospective, I would like to deprecate `set_debug_buffer` api and only focus on data sink for better structure.